sympy/solvers/solveset.py

```diff
@@ -539,23 +539,7 @@ def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        return Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
```

Comment:
[This line ](https://github.com/Shekharrajak/sympy/blob/master/sympy/solvers/solveset.py#L546) substituting `y` in place of `I*x` , it is okay for simple `exp` equations , but when we get some complicated equations like `exp(I*x/2)` ,then it is not able to substitute `y`.Most of the time because of this, [ here if condition](https://github.com/Shekharrajak/sympy/blob/master/sympy/solvers/solveset.py#L547) becomes `true` and returns `conditionSet`.So it is not able to solve many equation which can be solved easily.
If I use ` return solveset(f,symbol)` then now `f` is in `exp` form which is `complex` type and `solveset` is able to solve complicated `exp` equations.
So it gives correct output in reduced form which is desired.
@asmeurer  @aktech